timdex_dataset_api/metadata.py

```diff
@@ -436,36 +440,67 @@ def _create_current_records_view(self, conn: DuckDBPyConnection) -> None:
-        `timdex_record_id` if the `run_id` doesn't match, it's not the current version.
+        This metadata view includes only the most current version of each record in the
+        dataset.  With the metadata provided from this view, we can streamline data
+        retrievals in TIMDEXDataset read methods.
         """
         logger.info("creating view of current records metadata")
 
-        query = f"""
-        create or replace view metadata.current_records as
-        with ranked_records as (
+        conn.execute(
+            """
+            set temp_directory = '/tmp';
+            """
+        )
+
+        conn.execute(
+            """
+            -- create temp table with current records using CTEs
+            create or replace temp table temp.main.current_records as
+            with
+                -- CTE of run_timestamp for last source full run
+                cr_source_last_full as (
+                    select
+                        source,
+                        max(run_timestamp) as last_full_ts
+                    from metadata.records
+                    where run_type = 'full'
+                    group by source
+                ),
+
+                -- CTE of all records, per source, on or after last full run
+                cr_since_last_full as (
+                    select
+                        r.*
+                    from metadata.records r
+                    join cr_source_last_full f using (source)
+                    where r.run_timestamp >= f.last_full_ts
+                ),
+
+                -- CTE of records ranked by run_timestamp, with tie breaker
+                cr_ranked_records as (
+                    select
+                        r.*,
+                        row_number() over (
+                            partition by r.source, r.timdex_record_id
+                            order by
+                                r.run_timestamp desc nulls last,
+                                r.run_id desc nulls last,
+                                r.run_record_offset desc nulls last
+                        ) as rn
+                    from cr_since_last_full r
+                )
+
+            -- final select for current records (rn = 1)
             select
-                r.*,
-                row_number() over (
-                    partition by r.timdex_record_id
-                    order by r.run_timestamp desc
-                ) as rn
-            from metadata.records r
-            where r.run_timestamp >= (
-                select max(r2.run_timestamp)
-                from metadata.records r2
-                where r2.source = r.source
-                and r2.run_type = 'full'
-            )
+                * exclude (rn)
+            from cr_ranked_records
+            where rn = 1;
+
+            -- create view in metadata schema
+            create or replace view metadata.current_records as
```

Comment:
While we do now create a temporary table `temp.main.current_records`, we still maintain this `metadata.current_records` view as our common entrypoint of use.